doc/internal/man3/OSSL_TIME.pod

```diff
@@ -0,0 +1,90 @@
+=pod
+
+=head1 NAME
+
+OSSL_TIME, OSSL_TIME_SECOND, OSSL_TIME_NEVER,
+ossl_time_now, ossl_time_time_to_timeval, ossl_time_compare,
+ossl_time_add, ossl_time_subtract
+- times and durations
+
+=head1 SYNOPSIS
+
+ #include "internal/time.h"
+
+ typedef OSSL_TIME;
+
+ #define OSSL_TIME_SECOND
+ #define OSSL_TIME_NEVER
+
+ OSSL_TIME ossl_time_now(void);
```

Comment:
There's a valid question here as to what clock we want this to be. See clock_gettime(2). Since we are using this to schedule QUIC events we probably want something like `CLOCK_MONOTONIC` or `CLOCK_MONOTONIC_RAW` rather than `CLOCK_REALTIME`. Because of this we should document that this is not suitable for wall-clock time uses (e.g. checking certificate expiry).

The epoch used by this function probably doesn't need to be defined. It isn't for CLOCK_MONOTONIC.

The documented contract for this API can be deliberately vague. Something like: "This function returns the current time relative to an epoch which is undefined but unchanging for at least the duration of program execution. The time returned is monotonic and need not represent wall-clock time. The time returned by this function is useful for scheduling timeouts, deadlines and recurring events, but due to its undefined epoch and monotonic nature, is not suitable for other uses."

Reply:
+1 on the text which I've included.

Which clock we use will be an interesting discussion.  Currently, I've used what libssl uses for DTLS and I agree it's not ideal.  It's not a straightforward substitution, `gettimeofday(2)` is more widely available than `clock_gettime(2)` and doesn't require `-lrt` on the link line.  My guess is that we will want to support both falling back to the less accurate as required.

Various operating systems also expose higher resolution timers (Solaris, HP/UX and AIX from memory) which is an additional complexity I'd not to not dive into :grinning: